charts/metering-ansible-operator/templates/crds/meteringconfig.crd.yaml

```diff
@@ -159,6 +159,26 @@ spec:
                   type: string
                 secretName:
                   type: string
+              oneOf:
+              - required:
+                - enabled
+                properties:
+                  certificate:
+                    maxLength: 0
+                  key:
+                    maxLength: 0
```

Comment:
this kind of pattern (`maxLength: 0`) is necessary as `oneOf` only matches one valid sub-schema. In the case where I specify something like this (and don't specify the `maxLength: 0`):
```
  tls:
    enabled: false
    key: stub
```
This would be considered a valid `MeteringConfig` configuration, despite it being an invalid configuration of the TLS sub-schema. Alternatively, if I configured a `MeteringConfig` like this:
```
  tls:
    enabled: false
    key: stub
    certificate: ""
```
This configuration would still be considered valid, despite the `minLength: 1` property specified under the `spec.tls.certificate` property.